crates/merkle-inclusion/src/traits.rs

```diff
@@ -0,0 +1,57 @@
+use bellpepper_core::boolean::Boolean;
+use bellpepper_core::{ConstraintSystem, SynthesisError};
+use bellpepper_keccak::{keccak256, sha3};
+use ff::PrimeField;
+use sha3::{Keccak256, Sha3_256};
+
+/// A trait for implementing hash digest gadgets.
+///
+/// This trait defines the necessary methods for computing hash digests within constraint systems.
+pub trait GadgetDigest<E: PrimeField> {
+    /// Output size for our hasher, in bits.
+    const OUTPUT_SIZE: usize;
+
+    type OutOfCircuitHasher: digest::Digest;
+
+    /// Get output size of the hasher.
+    fn output_size() -> usize {
+        Self::OUTPUT_SIZE
```

Comment:
It would probably be useful to `debug_assert!` that this value is equal to `<Self::OutOfCircuitHasher as digest::Digest>::output_size()`.